src/sessionTypes/passwordGrantAuth.spec.js

```diff
@@ -0,0 +1,462 @@
+            },
+            auth: {
+              clientId: faker.internet.password()
+            }
+          }
+        };
+        password = faker.internet.password();
+        username = faker.internet.email();
+        expectedErrorMessage = faker.lorem.sentence();
+
+        authenticationSession = {
+          loginWithDefaultDirectory: this.sandbox
+            .stub()
+            .yields({ description: expectedErrorMessage })
+        };
+        this.sandbox
+          .stub(auth0, 'Authentication')
+          .returns(authenticationSession);
+
+        passwordGrantAuth = new PasswordGrantAuth(sdk);
+
+        promise = passwordGrantAuth.logIn(username, password);
+      });
+
+      it('returns a rejected promise with an error message', function() {
+        return expect(promise).to.be.eventually.rejectedWith(
+          expectedErrorMessage
+        );
+      });
+    });
+  });
+
+  describe('logOut', function() {
+    let passwordGrantAuth;
+    let promise;
+
+    beforeEach(function() {
+      sdk = {
+        config: {
+          audiences: {
+            contxtAuth: fixture.build('audience'),
+            facilities: fixture.build('audience')
+          },
+          auth: {
+            clientId: faker.internet.password()
+          }
+        }
+      };
+      authenticationSession = {
+        loginWithDefaultDirectory: this.sandbox.stub()
+      };
+      authentication = this.sandbox.stub(auth0, 'Authentication');
+
+      passwordGrantAuth = new PasswordGrantAuth(sdk);
+
+      promise = passwordGrantAuth.logOut();
+    });
+
+    it('clears out the session info', function() {
+      expect(passwordGrantAuth._sessionInfo).to.be.an('object');
+      expect(passwordGrantAuth._sessionInfo).to.be.empty;
+    });
+
+    it('returns a fulfilled promise with a success message', function() {
+      return promise.then((response) => {
+        expect(response).to.equal('Logout successful - session info cleared.');
+      });
+    });
+  });
+
+  describe('_getApiToken', function() {
+    beforeEach(function() {
+      sdk = {
+        config: {
+          audiences: {
+            contxtAuth: fixture.build('audience'),
+            facilities: fixture.build('audience')
+          },
+          auth: {
+            clientId: faker.internet.password()
+          }
+        }
+      };
+      authenticationSession = {
+        loginWithDefaultDirectory: this.sandbox.stub()
+      };
+      authentication = this.sandbox.stub(auth0, 'Authentication');
+    });
+
+    context('a successful request', function() {
+      context('when handling audiences with a client id', function() {
+        let accessToken;
+        let expectedApiToken;
+        let post;
+        let promise;
+        let saveSession;
+
+        beforeEach(function() {
+          accessToken = faker.internet.password();
+          expectedApiToken = faker.internet.password();
+
+          post = this.sandbox.stub(axios, 'post').callsFake(() => {
```

Comment:
Why not `post = this.sandbox.stub(axios, 'post').resolves({ data: { access_token: expectedApiToken }  });`?

Reply:
(this is very minor, im sorry)

Reply:
Good call. I think I had done it this way first and then started using `resolve`. I'll update this